src/mono/wasm/debugger/BrowserDebugProxy/ExpressionEvaluator.cs

```diff
@@ -436,9 +476,28 @@ private static object ConvertCSharpToJSType(object v, Type type)
                 return new { type = "number", value = v, description = Convert.ToDouble(v).ToString(CultureInfo.InvariantCulture) };
             if (v is JObject)
                 return v;
+            if (v is Array arr)
+            {
+                return resolver.CacheEvaluationResult(
+                    JObject.FromObject(
+                        new
+                        {
+                            type = "object",
+                            subtype = "array",
+                            value = new JArray(arr.Cast<object>().Select((val, idx) => JObject.FromObject(
+                                new
+                                {
+                                    value = ConvertCSharpToJSType(resolver, val, val.GetType()),
+                                    name = $"{idx}"
+                                }))),
+                            description = v.ToString(),
+                            className = type.ToString()
+                        }));
+            }
```

Comment:
I would suggest moving the bits to support methods returning "objects"/arrays to a follow up PR, so we can get rest of the changes right.

Reply:
The new PR: https://github.com/dotnet/runtime/pull/67095.